### ISSUE TITLE
Linter warnings bubble up as config formatting errors with confusing message

### ISSUE THREAD

**Issue by Bisaloo** [r-lib/lintr#2253]
See the following reprex package: https://github.com/Bisaloo/testlintrconfigerror/:

```dcf
linters: linters_with_defaults(
    backport_linter("2.10.0")
  )
```

The config file is correctly formatted but the `backport_linter()` setting will generate a warning:

https://github.com/r-lib/lintr/blob/1493c5e0bafc2483bea58a4f12404c68cf483e59/R/backport_linter.R#L116-L119

This warning is totally unrelated with the config formatting but will bubble up when config is read and produce a red herring message:

```r
lintr::lint_package()
```

> Error: Malformed config file, ensure it ends in a newline
>  Malformed config file, ensure it ends in a newline
>  It is not recommended to depend on an R version older than 3.0.0. Resetting 'r_version' to 3.0.0.

I looked again and again for the missing newline and couldn't figure out what was wrong until I realized the error came from the warning added at the end of the error message.
